src/ds/logger.h

```diff
@@ -511,17 +504,27 @@ namespace logger
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
 
-#define LOG_TRACE \
-  logger::config::ok(logger::TRACE) && \
-    logger::Out() == logger::LogLine(logger::TRACE, __FILE__, __LINE__)
-#define LOG_TRACE_FMT(s, ...) \
-  LOG_TRACE << fmt::format(FMT_STRING(s), ##__VA_ARGS__) << std::endl
-
-#define LOG_DEBUG \
-  logger::config::ok(logger::DEBUG) && \
-    logger::Out() == logger::LogLine(logger::DEBUG, __FILE__, __LINE__)
-#define LOG_DEBUG_FMT(s, ...) \
-  LOG_DEBUG << fmt::format(FMT_STRING(s), ##__VA_ARGS__) << std::endl
+#ifdef VERBOSE_LOGGING
+#  define LOG_TRACE \
+    logger::config::ok(logger::TRACE) && \
+      logger::Out() == logger::LogLine(logger::TRACE, __FILE__, __LINE__)
+#  define LOG_TRACE_FMT(s, ...) \
+    LOG_TRACE << fmt::format(FMT_STRING(s), ##__VA_ARGS__) << std::endl
+
+#  define LOG_DEBUG \
+    logger::config::ok(logger::DEBUG) && \
+      logger::Out() == logger::LogLine(logger::DEBUG, __FILE__, __LINE__)
+#  define LOG_DEBUG_FMT(s, ...) \
+    LOG_DEBUG << fmt::format(FMT_STRING(s), ##__VA_ARGS__) << std::endl
+#else
+// Without compile-time VERBOSE_LOGGING option, these logging macros are
+// compile-time nops (and cannot be enabled by accident or malice)
+#  define LOG_TRACE
```

Comment:
This will break if someone actually uses `LOG_TRACE` directly. The only place I found was in merklecpp, and that is currently disabled (see `src/node/history.h`: `// #include "merklecpp_trace.h"`). Given that you would only enable this if you also have `VERBOSE_LOGGING` enabled it should be fine.

Reply:
I think we can just drop that file. If we ever need tree-traces it's pretty easy to figure out how to enable them...